R/html_dependency.R

```diff
@@ -478,10 +496,37 @@ renderDependencies <- function(dependencies,
 
     # add scripts
     if (length(dep$script) > 0) {
+      if (length(dep$script) == 1) dep$script <- list(src = dep$script[[1]])
+
+      dep$script$src <- paste(
+        ' src="',
+        htmlEscape(hrefFilter(file.path(srcpath, encodeFunc(dep$script$src)))),
+        '"',
+        sep = ''
+      )
+
+      if (!is.null(dep$script$integrity)) {
+        dep$script$integrity <- paste(
+          ' integrity="', dep$script$integrity, '"',
+          sep = ''
+        )
+      }
+
+      if (!is.null(dep$script$crossorigin)) {
+        dep$script$crossorigin <- paste(
+          ' crossorigin="', htmlEscape(dep$script$crossorigin), '"',
+          sep = ''
+        )
+      } else {
+        dep$script$crossorigin <- ""
+      }
+
       html <- c(html, paste(
-        "<script src=\"",
-        htmlEscape(hrefFilter(file.path(srcpath, encodeFunc(dep$script)))),
-        "\"></script>",
+        "<script",
+        dep$script$src,
+        dep$script$integrity,
```

Comment:
Instead of the `if` statements up above, this could simply be:

```R
if (!is.null(dep$script$integrity)) paste0(' integrity="', dep$script$integrity, '"')
```

And similar for `crossorigin`.

Also, modifying the `dep` object in place and replacing the original values with these fragments of HTML doesn't feel right.

Reply:
> Also, modifying the dep object in place and replacing the original values with these fragments of HTML doesn't feel right.

Yeah I felt that too, I think I might refactor this out into a function anyway, though, because `<script>` isn't the only tag that can use these attributes, it's just the most important for `<script>`